### PR TITLE
Fix behavior of last 3 hours test mode switch (EXPOSUREAPP-3179)

### PR DESCRIPTION
**Pull request by d4rken**
Previously the days for which the last 3 hours were fetched, was the latest day in the servers index for a country, but if a day is not complete, the day will not be visible in the index. This PR fixes this behavior by artificially adding the one additional date (`LATEST_DAY_IN_INDEX+1DAY`).

## Testing
* Try it without this PR, check in the logs what the `KeyFileDownloader` downloads.
* Try with this PR, check that the actual last 3 hours are downloaded.

### CODE REVIEW

**Review by mlenkeit — COMMENTED** [review]
What if there are less than 3 hour packages available for the given date?

**Comment by mlenkeit** [comment]
> What if there are less than 3 hour packages available for the given date?

Just for the record: I think it's not required to fix this, as the 3 Hour mode is only used for testing, and it wouldn't work as the name suggests only between midnight and 2 am or so. I can live with that 😉

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/KeyFileDownloader.kt [hunk 1]
```diff
@@ -171,11 +172,21 @@ class KeyFileDownloader @Inject constructor(
...
+                    it.plus(lastDayInIndex.plusDays(1))
+                } else {
+                    it
+                }
+            }
+            Timber.tag(TAG).v("Day index with (fake) today entry: %s", indexWithToday)
+
+            indexWithToday.sortedDescending().mapNotNull { day ->
                 // Limit reached, return null (filtered out) instead of new CountryHours object
                 if (remainingItems <= 0) return@mapNotNull null
```
Comment (d4rken) [inline Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/KeyFileDownloader.kt:191]: @mlenkeit then we rollover to the previous day, see `remainingItems`.

Comment (d4rken) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/KeyFileDownloader.kt:191]: I'll push a PR for 1.6.x that changes to 24 hour mode and includes a unit-test for the rolloverbehavior.

**Review by BMItr — APPROVED** [review]

**Review by ralfgehrer — APPROVED** [review]

File: Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/KeyFileDownloader.kt (hunk 1 above)
Comment (ralfgehrer) [reply Corona-Warn-App/src/main/java/de/rki/coronawarnapp/diagnosiskeys/download/KeyFileDownloader.kt:191]: Max is fine with this change, too.

**Pull request merged**